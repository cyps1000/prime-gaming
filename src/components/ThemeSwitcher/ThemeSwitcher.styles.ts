/**
 *  Material UI Imports
 * @see https://material-ui.com/styles/basics/
 */
import { makeStyles } from "@material-ui/core/styles";

/**
 * Styles the component
 */
const useStyles = makeStyles((theme) => ({
  themeFix: {
    display: "flex",
    alignItems: "center",
    paddingLeft: 10,
    "& svg": {
      color: "#ffffff",
    },
  },
  switchFix: {
    paddingLeft: 10,
  },
  switchPrimary: {},
  switchTrack: {
    backgroundColor: theme.palette.common.white,
  },
}));

export { useStyles };
