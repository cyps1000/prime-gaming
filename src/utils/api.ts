import axios from "axios";

/**
 * Defines the hook props interface
 */
interface Config {
  mock: boolean;
}

export const getApiClient = (config: Config) => {
  const { mock } = config;

  /**
   * Handles getting the base api url
   */
  const getApiUrl = () => {
    if (process.env.NODE_ENV === "development") {
      if (mock) return "/";
      return process.env.REACT_APP_LOCAL_API;
    }
    return process.env.REACT_APP_STAGING_API;
  };

  const apiClient = axios.create({
    baseURL: getApiUrl(),
  });

  return { apiClient };
};
