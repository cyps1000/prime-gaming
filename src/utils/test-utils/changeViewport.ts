/**
 * External imports
 */
import mediaQuery from "css-mediaquery";

/**
 * Defines the BreakpointKey type
 */
type BreakpointKey = "xl" | "lg" | "md" | "sm" | "xs";

/**
 * Handles getting the width based on the breakpoint key provided
 */
const getWidthByThemeBreakpoints = (breakpointKey: BreakpointKey) => {
  switch (breakpointKey) {
    case "xl":
      return 1920;
    case "lg":
      return 1280;
    case "md":
      return 960;
    case "sm":
      return 600;
    case "xs":
      return 414;
    default:
      return 1920;
  }
};

/**
 * Handles changing the viewport
 */
export const changeViewport = (breakpointKey: BreakpointKey): any => {
  let width: number = getWidthByThemeBreakpoints(breakpointKey);

  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width: width }),
      addListener: () => {},
      removeListener: () => {},
    };
  };
};
