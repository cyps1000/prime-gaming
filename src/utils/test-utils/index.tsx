import React, { ReactElement } from "react";
/**
 * External Imports
 */
import { MemoryRouter } from "react-router-dom";

/**
 * Imports testing utils
 */
import { render, RenderOptions } from "@testing-library/react";

/**
 * Imports providers
 */

import { ThemeProvider, LanguageProvider } from "../../hooks";
import Providers from "../../components/Providers";

/**
 * Defines the Wrapper component
 */
const Wrapper: React.FC = ({ children }) => {
  return (
    <ThemeProvider>
      <LanguageProvider>
        <Providers>
          <MemoryRouter>{children}</MemoryRouter>
        </Providers>
      </LanguageProvider>
    </ThemeProvider>
  );
};

/**
 * Defines the custom render
 */
const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options });

/**
 * Re-exports everything from the testing library
 */
export * from "@testing-library/react";

export * from "./changeViewport";
export * from "./mocks";
export { default as userEvent } from "@testing-library/user-event";
export { customRender as render };
