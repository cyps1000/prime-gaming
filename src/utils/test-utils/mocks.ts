export const mockHistoryPush = jest.fn();
