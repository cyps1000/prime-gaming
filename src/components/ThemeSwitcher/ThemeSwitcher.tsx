/**
 * Imports Material UI Components
 */
import Switch from "@material-ui/core/Switch";
import Brightness4Icon from "@material-ui/icons/Brightness4";
import Brightness7Icon from "@material-ui/icons/Brightness7";

/**
 * Imports the component styles
 */
import { useStyles } from "./ThemeSwitcher.styles";

/**
 * Imports Hooks
 */
import { useTheme } from "../../hooks";

/**
 * Defines the props interface
 */
export interface ThemeSwitcherProps {
  text?: string;
}

/**
 * Displays the component
 */
const ThemeSwitcher: React.FC<ThemeSwitcherProps> = (props) => {
  const { text } = props;

  /**
   * Gets the active theme and theme changer
   */
  const { activeTheme, changeTheme } = useTheme();

  /**
   * Gets the component styles
   */
  const classes = useStyles();

  /**
   * Handles changing the theme
   */
  const handleThemeChange = () => {
    changeTheme(activeTheme === "dark-theme" ? "light-theme" : "dark-theme");
  };

  return (
    <div className={classes.themeFix}>
      <Brightness7Icon />
      <Switch
        checked={activeTheme === "dark-theme"}
        onChange={handleThemeChange}
        color="secondary"
        name="checkedB"
        className={classes.switchFix}
        classes={{
          colorPrimary: classes.switchPrimary,
        }}
        inputProps={{ "aria-label": "primary checkbox" }}
      />
      <Brightness4Icon />
    </div>
  );
};

export default ThemeSwitcher;
