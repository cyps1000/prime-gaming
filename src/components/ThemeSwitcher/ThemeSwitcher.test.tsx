/**
 * @see https://testing-library.com/docs/react-testing-library/intro
 * @see https://www.robinwieruch.de/react-testing-library
 * @see https://www.smashingmagazine.com/2020/07/react-apps-testing-library/
 */

/**
 * Imports test utils
 */
import { render } from "../../utils/test-utils";

/**
 * External Imports
 */
import pretty from "pretty";

/**
 * Imports component
 */
import ThemeSwitcher from "./ThemeSwitcher";

/**
 * ThemeSwitcher rendering tests
 */
describe("ThemeSwitcher Rendering Tests", () => {
  it("renders the component without errors", () => {
    render(<ThemeSwitcher />);
  });
});

/**
 * ThemeSwitcher logic tests
 */
describe("ThemeSwitcher Logic Tests", () => {
  it("changes the theme of the app when switching", () => {
    const { getByRole } = render(<ThemeSwitcher />);

    expect(getByRole("checkbox")).toHaveProperty("value", "dark-theme");
    expect(getByRole("checkbox")).toHaveProperty("checked", true);

    getByRole("checkbox").click();

    expect(getByRole("checkbox")).toHaveProperty("value", "light-theme");
    expect(getByRole("checkbox")).toHaveProperty("checked", false);

    getByRole("checkbox").click();

    expect(getByRole("checkbox")).toHaveProperty("value", "dark-theme");
    expect(getByRole("checkbox")).toHaveProperty("checked", true);
  });
});

/**
 * ThemeSwitcher snapshot test
 */
describe("ThemeSwitcher Snapshot Test", () => {
  it("passes the snapshot test", () => {
    const { container } = render(<ThemeSwitcher />);

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
