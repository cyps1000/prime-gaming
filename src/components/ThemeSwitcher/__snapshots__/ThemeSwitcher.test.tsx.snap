// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitcher Snapshot Test passes the snapshot test 1`] = `
"<div class=\\"makeStyles-container-33\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\">
    <path d=\\"M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm0-10c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4z\\"></path>
  </svg><span class=\\"MuiSwitch-root makeStyles-switch-34\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-37 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-38 Mui-checked\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input class=\\"PrivateSwitchBase-input-40 MuiSwitch-input\\" name=\\"checkedB\\" type=\\"checkbox\\" aria-label=\\"primary checkbox\\" value=\\"dark-theme\\" checked=\\"\\"><span class=\\"MuiSwitch-thumb\\"></span></span><span class=\\"MuiTouchRipple-root\\"></span></span><span class=\\"MuiSwitch-track\\"></span></span><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\">
    <path d=\\"M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z\\"></path>
  </svg></div>"
`;
